src-electron/db/query-loader.js

```diff
@@ -938,54 +938,80 @@ async function insertAtomics(db, packageId, data) {
  * @param {*} context - The context containing the mandatory device type to check against.
  * @returns A promise resolved with the result of the database insert operation.
  */
-function insertEndpointComposition(db, composition, context) {
-  if (parseInt(context.mandatoryDeviceTypes, 16) === composition.code) {
-    return dbApi.dbInsert(
-      db,
-      'INSERT INTO ENDPOINT_COMPOSITION (TYPE, CODE) VALUES (?, ?)',
-      [dbEnum.composition.mandatoryEndpoint, composition.code],
-    )
-  } else {
-    return dbApi.dbInsert(
-      db,
-      'INSERT INTO ENDPOINT_COMPOSITION (TYPE, CODE) VALUES (?, ?)',
-      [composition.compositionType, composition.code],
-    )
+async function insertEndpointComposition(db, composition, context) {
+  try {
+    if (parseInt(context.mandatoryDeviceTypes, 16) === composition.code) {
+      return await dbApi.dbInsert(
+        db,
+        'INSERT INTO ENDPOINT_COMPOSITION (TYPE, CODE) VALUES (?, ?)',
+        [dbEnum.composition.mandatoryEndpoint, composition.code],
+      )
+    } else {
+      return await dbApi.dbInsert(
+        db,
+        'INSERT INTO ENDPOINT_COMPOSITION (TYPE, CODE) VALUES (?, ?)',
+        [composition.compositionType, composition.code],
+      )
+    }
+  } catch (error) {
+    console.error('Error inserting endpoint composition:', error)
+    throw error // Re-throw the error after logging it
```

Comment:
These seem needless. You're already getting an error. You are also not "logging" it. You're just printing it out.
We have our "log" module if you want to log stuff.

Reply:
Ok is this an issue?

Reply:
Yes. Put it in the log. We don't want to use `console.error`